### PR TITLE
Rename architecture_implementation to TTDevice

### PR DESCRIPTION
**Pull request by broskoTT**
### Issue
First step in #98 

### Description
For more information consult https://docs.google.com/drawings/d/1-m1azdsBqMA0A6ATYRMfkhyeuOJuGCEI62N5a96LXj0/edit
The following change will implement further changes to set TTDevice own PCIDevice, rather than other way around.

### List of the changes
- Renamed architecture_implementation to tt_device.h and class TTDevice
- Renamed bh, wh and gs accordingly.
- No functional changes

### Testing
Ci tests, code builds

### API Changes
There are no API changes in this PR.

### CODE REVIEW

File: src/firmware/riscv/grayskull/eth_interface.h [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 #pragma once
 
+#include <cstdint>
```
Comment (pjanevskiTT) [inline src/firmware/riscv/grayskull/eth_interface.h:7]: why?

**Review by pjanevskiTT — APPROVED** [review]
Nice!

File: src/firmware/riscv/grayskull/eth_interface.h (hunk 1 above)
Comment (broskoTT) [reply src/firmware/riscv/grayskull/eth_interface.h:7]: uint32_t definition

File: device/api/umd/device/tt_device_blackhole.h [hunk 2]
```diff
@@ -145,7 +145,7 @@ static constexpr uint32_t MSG_TYPE_SETUP_IATU_FOR_PEER_TO_PEER = 0x97;
 
 }  // namespace blackhole
 
-class blackhole_implementation : public architecture_implementation {
+class BlackholeTTDevice : public TTDevice {
```
Comment (broskoTT) [inline device/api/umd/device/tt_device_blackhole.h:148]: Think about file names for these TTDevice

Comment (broskoTT) [reply device/api/umd/device/tt_device_blackhole.h:148]: Will rename to api/umd/device/tt_device/grayskull_tt_device.h

**Comment by broskoTT** [comment]
Deprecated by #359

**Pull request closed without merging**